apis/apiextensions/v1/composition_types.go

```diff
@@ -552,11 +562,23 @@ type ConnectionDetail struct {
 	// +optional
 	Name *string `json:"name,omitempty"`
 
+	// Type sets the connection detail fetching behaviour to be used. Each connection detail type may require
+	// its' own fields to be set on the ConnectionDetail object.
+	// +optional
+	// +kubebuilder:validation:Enum=FromConnectionSecretKey;FromResourceFieldPath;Value
+	// +kubebuilder:default=FromConnectionSecretKey
+	Type ConnectionDetailType `json:"type,omitempty"`
```

Comment:
I based this on the PatchType field which is not a pointer

```
// Type sets the patching behaviour to be used. Each patch type may require
// its' own fields to be set on the Patch object.
// +optional
// +kubebuilder:validation:Enum=FromCompositeFieldPath;PatchSet;ToCompositeFieldPath
// +kubebuilder:default=FromCompositeFieldPath
Type PatchType `json:"type,omitempty"`
```

Reply:
Got it. I feel good about these being non-pointers since in practice the API server will make sure they always have a value set thanks to the `default`.